sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -552,4 +577,155 @@ class AdaptiveQueryExecSuite
       spark.sparkContext.removeSparkListener(listener)
     }
   }
+
+  test("adaptive skew join both in left and right for inner join ") {
+    withSQLConf(
+      SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1",
+      SQLConf.ADAPTIVE_EXECUTION_SKEWED_PARTITION_FACTOR.key -> "1",
+      SQLConf.ADAPTIVE_EXECUTION_SKEWED_PARTITION_SIZE_THRESHOLD.key -> "100",
+      SQLConf.SHUFFLE_TARGET_POSTSHUFFLE_INPUT_SIZE.key -> "2000") {
+      val (plan, adaptivePlan) = runAdaptiveAndVerifyResult(
+        "SELECT * FROM skewData1 join skewData2 ON key1 = key2")
+      val smj = findTopLevelSortMergeJoin(plan)
+      assert(smj.size == 1)
+      // left stats: [4403, 0, 1927, 1927, 1927]
```

Comment:
I'm a little surprised that partition 1 is skewed. Isn't `key1` uniformly distributed?

Reply:
I think it's because `ADAPTIVE_EXECUTION_SKEWED_PARTITION_SIZE_THRESHOLD` is set to 1 but it feels like cheating.

Reply:
It may be caused by the calculation in `ShuffleExchangeExec`. And also update the test data.

Reply:
Sorry, I mean `ADAPTIVE_EXECUTION_SKEWED_PARTITION_FACTOR`